python/ray/data/_internal/execution/autoscaler/autoscaling_actor_pool.py

```diff
@@ -61,46 +89,20 @@ def num_free_task_slots(self) -> int:
         """
         return (
             self.max_tasks_in_flight_per_actor() * self.num_running_actors()
-            - self.current_in_flight_tasks()
+            - self.num_tasks_in_flight()
         )
 
     @abstractmethod
-    def can_scale_down(self):
-        ...
-
-    @abstractmethod
-    def scale_up(self, num_actors: int, *, reason: Optional[str] = None) -> int:
-        """Request the actor pool to scale up by the given number of actors.
-
-        The number of actually added actors may be less than the requested
-        number.
-
-        Args:
-            num_actors: Number of additional actors to be added to the pool
-            reason: (Optional) Reason for action
-
-        Returns:
-            The number of actors actually added.
-        """
+    def scale(self, config: ActorPoolScalingRequest):
+        """Applies autoscaling action"""
         ...
 
     @abstractmethod
-    def scale_down(self, num_actors: int, *, reason: Optional[str] = None) -> int:
-        """Request actor pool to scale down by the given number of actors.
-
-        The number of actually removed actors may be less than the requested
-        number.
-
-        Args:
-            num_actors: Number of additional actors to be removed from the pool
-            reason: (Optional) Reason for action
-
-        Returns:
-            The number of actors actually removed.
-        """
+    def per_actor_resource_usage(self) -> ExecutionResources:
+        """Per actor resource usage."""
         ...
 
     @abstractmethod
-    def per_actor_resource_usage(self) -> ExecutionResources:
-        """Per actor resource usage."""
+    def get_pool_util(self) -> float:
+        """Calculate the utilization of the given actor pool."""
         ...
```

Comment:
Should this be the responsibility of the autoscaler? You can compute this just using public actor pool methods, and I think the actor pool interface is already really bloated

Reply:
Yeah, i went back and forth on this. Ultimately decided that the scope of AP should include defining what util actually means (in terms of # of actors used, concurrency slots or what not) as these are intrinsic actor details. Autoscaler on the other hand operates at a higher level (if util > threshold then upscale for ex)